### PR TITLE
HPCC-25669 Update IHttpClient to take a CTxSummary

### PR DESCRIPTION
**Pull request by asselitx**
This is preparation for enabling the HttpClient to log TxSummary lines
specific to its activity.

Signed-off-by: Terrence Asselin <terrence.asselin@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [x] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/bindings/http/client/httpclient.hpp [hunk 1]
```diff
@@ -41,6 +41,7 @@ interface IHttpClient : extends ITransportClient
 
     virtual void setConnectTimeOutMs(unsigned timeout) = 0;
     virtual void setTimeOut(unsigned int timeout) = 0;
+    virtual void setTxSummary(CTxSummary* txSummary) = 0;
```
Comment (timothyklemm) [inline esp/bindings/http/client/httpclient.hpp:44]: Is the visibility of CTxSummary guaranteed here? I might forward declare the class just to be safe.

**Review by timothyklemm — COMMENTED** [review]
This is the direction I was expecting it to go. It's too bad it's not ready to merge.

File: esp/bindings/http/client/httpclient.hpp (hunk 1 above)
Comment (asselitx) [reply esp/bindings/http/client/httpclient.hpp:44]: There is already a forward declaration in esp.hpp included in soapesp.hpp which is included in httpclient.hpp.

**Review by timothyklemm — APPROVED** [review]
The code changes are accepted. However, they relate to HPCC-25669 which is a subtask of HPCC-25644.

**Comment by richardkchapman** [comment]
Should the commit title be modified to refer to the correct Jira?

**Comment by timothyklemm** [comment]
Yes, the title should be changed, and the correct Jira task should be updated. This piece of the larger larger task was separated because it is the only part of the larger task that is needed by the external security managers I'm working on which require a separate IHttpClient implementation.

**Pull request merged**